src/Polly.Core/Simmy/Latency/LatencyChaosStrategy.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Polly.Telemetry;
+
+namespace Polly.Simmy.Latency;
+
+#pragma warning disable S3928 // Custom ArgumentNullException message
+
+internal sealed class LatencyChaosStrategy : MonkeyStrategy
+{
+    private readonly TimeProvider _timeProvider;
+    private readonly ResilienceStrategyTelemetry _telemetry;
+
+    public LatencyChaosStrategy(
+        LatencyStrategyOptions options,
+        TimeProvider timeProvider,
+        ResilienceStrategyTelemetry telemetry)
+        : base(options)
+    {
+        Latency = options.Latency;
+        LatencyGenerator = options.LatencyGenerator is not null ? options.LatencyGenerator : (_) => new(options.Latency);
```

Comment:
Is this creating a new value in the Lambda on every invocation? If it is, could you just compute it once and return that every time?

Reply:
It's just when the strategy is created and only when the generator isn't specified.